rotkehlchen/db/dbhandler.py

```diff
@@ -3022,26 +3020,31 @@ def delete_bitcoin_xpub(
         """
         write_cursor.execute(
             'SELECT COUNT(*) FROM xpubs WHERE xpub=? AND derivation_path IS ? AND blockchain=?;',
-            (xpub_data.xpub.xpub, xpub_data.serialize_derivation_path_for_db(), blockchain.value),
+            (
+                xpub_data.xpub.xpub,
+                xpub_data.serialize_derivation_path_for_db(),
+                xpub_data.blockchain.value,
+            ),
         )
         if write_cursor.fetchone()[0] == 0:
             derivation_str = (
                 'no derivation path' if xpub_data.derivation_path is None else
                 f'derivation path {xpub_data.derivation_path}'
             )
             raise InputError(
-                f'Tried to remove non existing xpub {xpub_data.xpub.xpub} for {blockchain.value} '
-                f'with {derivation_str}',
+                f'Tried to remove non existing xpub {xpub_data.xpub.xpub} '
+                f'for {xpub_data.blockchain.value} with {derivation_str}',
             )
 
         # Delete the tag mappings for all derived addresses
         write_cursor.execute(
             'DELETE FROM tag_mappings WHERE '
             'object_reference IN ('
-            'SELECT address from xpub_mappings WHERE xpub=? and derivation_path IS ?);',
+            'SELECT address from xpub_mappings WHERE xpub=? AND derivation_path IS ? AND blockchain=?);',  # noqa: E501
```

Comment:
Aha so this is why so far if one had a BTC xpub he also appeared to have BCH accounts with 0 value? Which probably makes sense as discussed in discord, but we would need to have a change in the approach that's more generic to have this work across the different BTC chains, EVM chains etc.